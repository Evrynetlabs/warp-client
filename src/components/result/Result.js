import React, { useState } from 'react'
import { Card, Container, Row, Col, Button } from 'react-bootstrap'
import PropTypes from 'prop-types'
import { useGetPageTypeItems } from 'Components/result/resultHooks'
import 'Components/result/result.scss'
import classNames from 'classnames'
import { toCurrency } from '@/utils/format'
import { CopyToClipboard } from 'react-copy-to-clipboard'

const Result = (props) => {
  const [stylesMain] = useState('Result')
  const {
    amount,
    asset: { code, decimal },
    src,
    dest,
    txHashes,
    removeResult,
  } = props
  const { items } = useGetPageTypeItems(props)
  const SrcIcon = items.chain.src
  const DestIcon = items.chain.dest

  return (
    <Container className={classNames(stylesMain, 'p-0')}>
      <Row>
        <Col className={`${stylesMain}__content`}>
          <Card className={`${stylesMain}__content__card`}>
            <Card.Body className="p-0">
              <div
                className={classNames({
                  [`${stylesMain}__content__body`]: true,
                })}
              >
                <Container className="p-0">
                  <Row
                    className={classNames(
                      'm-0',
                      `${stylesMain}__content__body__row`,
                    )}
                  >
                    <Col
                      className={classNames(
                        `${stylesMain}__content__body__status`,
                      )}
                    >
                      <span>{items.title}</span>
                    </Col>
                  </Row>
                  <Row
                    className={classNames(
                      'm-0',
                      `${stylesMain}__content__body__row`,
                      'justify-content-start',
                    )}
                  >
                    <Col
                      className={classNames({
                        [`${stylesMain}__content__body__title`]: true,
                      })}
                    >
                      <span
                        className={classNames(
                          'label--dark',
                          'text-format-title',
                          `${stylesMain}__content__body__title__text`,
                        )}
                      >
                        From
                      </span>
                      <SrcIcon></SrcIcon>
                    </Col>
                    <Col
                      className={classNames(
                        `${stylesMain}__content__body__title`,
                        'text-right',
                      )}
                    >
                      <span
                        className={classNames(
                          'label--dark',
                          'text-format-title',
                          `${stylesMain}__content__body__title__text`,
                        )}
                      >
                        To
                      </span>
                      <DestIcon></DestIcon>
                    </Col>
                  </Row>
                  <Row
                    className={classNames(
                      'm-0',
                      `${stylesMain}__content__body__row`,
                      'justify-content-start',
                    )}
                  >
                    <Col
                      className={classNames({
                        [`${stylesMain}__content__body__account`]: true,
                      })}
                    >
                      <div className="text-left ellipsis">
                        <span
                          className={classNames(
                            `${stylesMain}__content__body__account__text`,
                            'text-format-input',
                          )}
                        >
                          {src}
                        </span>
                      </div>
                    </Col>
                    <Col className={`${stylesMain}__content__body__swap`}>
                      <i className="fas fa-arrow-right"></i>
                    </Col>
                    <Col
                      className={classNames({
                        [`${stylesMain}__content__body__account`]: true,
                      })}
                    >
                      <div className="text-right ellipsis">
                        <span
                          className={classNames(
                            `${stylesMain}__content__body__account__text`,
                            'text-format-input',
                          )}
                        >
                          {dest}
                        </span>
                      </div>
                    </Col>
                  </Row>
                  <Row
                    className={classNames(
                      'm-0',
                      `${stylesMain}__content__body__row`,
                      'justify-content-start',
                    )}
                  >
                    <Col
                      className={classNames({
                        [`${stylesMain}__content__body__title`]: true,
                      })}
                    >
                      <span
                        className={classNames(
                          'label--dark',
                          'text-format-title',
                          `${stylesMain}__content__body__title__text`,
                        )}
                      >
                        Amount
                      </span>
                    </Col>
                  </Row>
                  <Row
                    className={classNames(
                      'm-0',
                      `${stylesMain}__content__body__row`,
                    )}
                  >
                    <Col
                      className={classNames(
                        `${stylesMain}__content__body__amount`,
                        'text-format-title-bold',
                      )}
                    >
                      <span>{`${toCurrency(amount, decimal)} ${code}`}</span>
                    </Col>
                  </Row>
                  <hr />
                  <Row
                    className={classNames(
                      'm-0',
                      `${stylesMain}__content__body__row`,
                    )}
                  >
                    <Col
                      className={classNames({
                        [`${stylesMain}__content__hash`]: true,
                      })}
                    >
                      <Card>
                        <Card.Body
                          className={classNames({
                            [`${stylesMain}__content__hash__content`]: true,
                          })}
                        >
                          <span
                            className={classNames(
                              'text-format-label',
                              'd-block',
                            )}
                          >
                            For Stellar
                          </span>
                          <Container className="m-0">
                            <Row className="p-0 justify-content-between">
                              <Col>
                                <span
                                  className={classNames('text-format-input')}
                                >
                                  {txHashes.state.stellar}
                                </span>
                              </Col>
                              <Col className="text-right">
                                <CopyToClipboard text={txHashes.state.stellar}>
                                  <Button>Copy</Button>
                                </CopyToClipboard>
                              </Col>
                            </Row>
                          </Container>
                        </Card.Body>
                      </Card>
                    </Col>
                  </Row>
                  <Row
                    className={classNames(
                      'm-0',
                      `${stylesMain}__content__body__row`,
                    )}
                  >
                    <Col
                      className={classNames({
                        [`${stylesMain}__content__hash`]: true,
                      })}
                    >
                      <Card>
                        <Card.Body
                          className={classNames({
                            [`${stylesMain}__content__hash__content`]: true,
                          })}
                        >
                          <span
                            className={classNames(
                              'text-format-label',
                              'd-block',
                            )}
                          >
                            For E^vrynet
                          </span>
                          <Container className="m-0">
                            <Row className="p-0 justify-content-between">
                              <Col>
                                <span
                                  className={classNames('text-format-input')}
                                >
                                  {txHashes.state.evrynet}
                                </span>
                              </Col>
                              <Col className="text-right">
                                <CopyToClipboard text={txHashes.state.evrynet}>
                                  <Button>Copy</Button>
                                </CopyToClipboard>
                              </Col>
                            </Row>
                          </Container>
                        </Card.Body>
                      </Card>
                    </Col>
                  </Row>
                  <Row
                    className={classNames(
                      'm-0',
                      `${stylesMain}__content__body__row`,
                    )}
                  >
                    <Col className={`${stylesMain}__close`}>
                      <Button
                        onClick={() => {
                          removeResult()
                        }}
                        variant="secondary"
                        className={classNames(`${stylesMain}__close__btn`)}
                      >
                        Done
                      </Button>
                    </Col>
                  </Row>
                </Container>
              </div>
            </Card.Body>
          </Card>
        </Col>
      </Row>
    </Container>
  )
}

Result.propTypes = {
  push: PropTypes.func.isRequired,
  amount: PropTypes.string.isRequired,
  asset: PropTypes.shape({
    code: PropTypes.string,
    decimal: PropTypes.number,
  }).isRequired,
  src: PropTypes.string.isRequired,
  dest: PropTypes.string.isRequired,
  txHashes: PropTypes.shape({
    state: PropTypes.shape({
      stellar: PropTypes.string,
      evrynet: PropTypes.string,
    }),
    error: PropTypes.object,
  }).isRequired,
  removeResult: PropTypes.func.isRequired,
  isToEvrynet: PropTypes.bool.isRequired,
}

export default Result
