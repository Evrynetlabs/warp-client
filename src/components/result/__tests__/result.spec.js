import React from 'react'
import { shallow } from 'enzyme'
import Result from 'Components/result/Result.js'

describe('Result', () => {
  const mockSuccessHashes = {
    state: {
      stellar: 'foo',
      evrynet: 'bar',
    },
    error: null,
  }
  const mockFailedHashes = {
    state: null,
    error: new Error('this is an error'),
  }
  const mock = {
    push: jest.fn(),
    match: {},
    location: {
      state: {
        isToEvrynet: true,
        amount: '1000',
        asset: { code: 'EVRY', decimal: 7 },
        src: 'foo',
        dest: 'bar',
      },
    },
    history: {},
  }

  describe('When the transfer process is success', () => {
    describe('When isToEverynet', () => {
      it('should render correspondingly', () => {
        const component = shallow(
          <Result
            push={mock.push}
            match={mock.match}
            location={{
              state: {
                ...mock.location.state,
                txHashes: mockSuccessHashes,
              },
            }}
            history={mock.history}
          ></Result>,
        )
        expect(component).toMatchSnapshot()
      })
    })

    describe('When isToStellar', () => {
      it('should render correspondingly', () => {
        const component = shallow(
          <Result
            push={mock.push}
            match={mock.match}
            location={{
              state: {
                ...mock.location.state,
                txHashes: mockSuccessHashes,
              },
            }}
            history={mock.history}
          ></Result>,
        )
        expect(component).toMatchSnapshot()
      })
    })
  })

  describe('When the transfer process is failed', () => {
    describe('When isToEverynet', () => {
      it('should render correspondingly', () => {
        const component = shallow(
          <Result
            push={mock.push}
            match={mock.match}
            location={{
              state: {
                ...mock.location.state,
                txHashes: mockFailedHashes,
              },
            }}
            history={mock.history}
          ></Result>,
        )
        expect(component).toMatchSnapshot()
      })
    })

    describe('When isToStellar', () => {
      it('should render correspondingly', () => {
        const component = shallow(
          <Result
            push={mock.push}
            match={mock.match}
            location={{
              state: {
                ...mock.location.state,
                txHashes: mockFailedHashes,
              },
            }}
            history={mock.history}
          ></Result>,
        )
        expect(component).toMatchSnapshot()
      })
    })
  })
})
