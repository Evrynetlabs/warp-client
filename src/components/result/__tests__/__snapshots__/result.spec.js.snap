// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Result When the transfer process is failed When isToEverynet should render correspondingly 1`] = `
<Container
  className="Result"
  fluid={false}
>
  <ForwardRef
    noGutters={false}
  >
    <Col
      className="Result__content"
    >
      <Card
        body={false}
      >
        <CardTitle>
          <div
            className="text-center Result__content__title--failed"
          >
            <i
              className="fas fa-times-circle"
            />
          </div>
        </CardTitle>
        <CardBody
          className="p-0"
        >
          <div
            className="Result__content__body"
          >
            <Container
              fluid={false}
            >
              <ForwardRef
                noGutters={false}
              >
                <Col
                  className="Result__content__body__amount"
                >
                  <h1>
                    1,000 EVRY
                  </h1>
                </Col>
              </ForwardRef>
              <ForwardRef
                noGutters={false}
              >
                <Col
                  className="Result__content__body__status--failed"
                >
                  <h4>
                    TRANSACTION FAILED
                  </h4>
                </Col>
              </ForwardRef>
              <ForwardRef
                noGutters={false}
              >
                <Col
                  className="Result__content__body__account"
                  sm={5}
                  xs={12}
                >
                  <div
                    className="text-left"
                  >
                    <span
                      className="label--dark"
                    >
                      From
                    </span>
                    <Card
                      body={false}
                    >
                      <CardBody>
                        <Container
                          fluid={false}
                        >
                          <ForwardRef
                            noGutters={false}
                          >
                            <span>
                              Stellar
                            </span>
                          </ForwardRef>
                          <ForwardRef
                            noGutters={false}
                          >
                            <span
                              className="Result__content__body__account__text"
                            >
                              Account Number: 
                              foo
                            </span>
                          </ForwardRef>
                        </Container>
                      </CardBody>
                    </Card>
                  </div>
                </Col>
                <Col
                  className="Result__content__body__swap"
                >
                  <i
                    className="fas fa-arrow-right"
                  />
                </Col>
                <Col
                  className="Result__content__body__account"
                  sm={5}
                  xs={12}
                >
                  <div
                    className="text-left"
                  >
                    <span
                      className="label--dark"
                    >
                      To
                    </span>
                    <Card
                      body={false}
                    >
                      <CardBody>
                        <Container
                          fluid={false}
                        >
                          <ForwardRef
                            noGutters={false}
                          >
                            <span>
                              Evrynet
                            </span>
                          </ForwardRef>
                          <ForwardRef
                            noGutters={false}
                          >
                            <span
                              className="Result__content__body__account__text"
                            >
                              Account Number: 
                              bar
                            </span>
                          </ForwardRef>
                        </Container>
                      </CardBody>
                    </Card>
                  </div>
                </Col>
              </ForwardRef>
            </Container>
          </div>
        </CardBody>
        <CardFooter>
          <div
            className="Result__content__footer"
          >
            <Container
              fluid={false}
            >
              <ForwardRef
                noGutters={false}
              >
                <Col
                  className="Result__content__footer__content"
                  xs={12}
                >
                  <Card
                    body={false}
                  >
                    <CardBody
                      className="Result__content__footer__content__body"
                    >
                      Error: 
                      Error: this is an error
                    </CardBody>
                  </Card>
                </Col>
              </ForwardRef>
            </Container>
          </div>
        </CardFooter>
      </Card>
    </Col>
  </ForwardRef>
  <ForwardRef
    noGutters={false}
  >
    <Col
      className="Result__redirect"
    >
      <Button
        active={false}
        disabled={false}
        onClick={[Function]}
        type="button"
        variant="primary"
      >
        Back to Home Page
      </Button>
    </Col>
  </ForwardRef>
</Container>
`;

exports[`Result When the transfer process is failed When isToStellar should render correspondingly 1`] = `
<Container
  className="Result"
  fluid={false}
>
  <ForwardRef
    noGutters={false}
  >
    <Col
      className="Result__content"
    >
      <Card
        body={false}
      >
        <CardTitle>
          <div
            className="text-center Result__content__title--failed"
          >
            <i
              className="fas fa-times-circle"
            />
          </div>
        </CardTitle>
        <CardBody
          className="p-0"
        >
          <div
            className="Result__content__body"
          >
            <Container
              fluid={false}
            >
              <ForwardRef
                noGutters={false}
              >
                <Col
                  className="Result__content__body__amount"
                >
                  <h1>
                    1,000 EVRY
                  </h1>
                </Col>
              </ForwardRef>
              <ForwardRef
                noGutters={false}
              >
                <Col
                  className="Result__content__body__status--failed"
                >
                  <h4>
                    TRANSACTION FAILED
                  </h4>
                </Col>
              </ForwardRef>
              <ForwardRef
                noGutters={false}
              >
                <Col
                  className="Result__content__body__account"
                  sm={5}
                  xs={12}
                >
                  <div
                    className="text-left"
                  >
                    <span
                      className="label--dark"
                    >
                      From
                    </span>
                    <Card
                      body={false}
                    >
                      <CardBody>
                        <Container
                          fluid={false}
                        >
                          <ForwardRef
                            noGutters={false}
                          >
                            <span>
                              Stellar
                            </span>
                          </ForwardRef>
                          <ForwardRef
                            noGutters={false}
                          >
                            <span
                              className="Result__content__body__account__text"
                            >
                              Account Number: 
                              foo
                            </span>
                          </ForwardRef>
                        </Container>
                      </CardBody>
                    </Card>
                  </div>
                </Col>
                <Col
                  className="Result__content__body__swap"
                >
                  <i
                    className="fas fa-arrow-right"
                  />
                </Col>
                <Col
                  className="Result__content__body__account"
                  sm={5}
                  xs={12}
                >
                  <div
                    className="text-left"
                  >
                    <span
                      className="label--dark"
                    >
                      To
                    </span>
                    <Card
                      body={false}
                    >
                      <CardBody>
                        <Container
                          fluid={false}
                        >
                          <ForwardRef
                            noGutters={false}
                          >
                            <span>
                              Evrynet
                            </span>
                          </ForwardRef>
                          <ForwardRef
                            noGutters={false}
                          >
                            <span
                              className="Result__content__body__account__text"
                            >
                              Account Number: 
                              bar
                            </span>
                          </ForwardRef>
                        </Container>
                      </CardBody>
                    </Card>
                  </div>
                </Col>
              </ForwardRef>
            </Container>
          </div>
        </CardBody>
        <CardFooter>
          <div
            className="Result__content__footer"
          >
            <Container
              fluid={false}
            >
              <ForwardRef
                noGutters={false}
              >
                <Col
                  className="Result__content__footer__content"
                  xs={12}
                >
                  <Card
                    body={false}
                  >
                    <CardBody
                      className="Result__content__footer__content__body"
                    >
                      Error: 
                      Error: this is an error
                    </CardBody>
                  </Card>
                </Col>
              </ForwardRef>
            </Container>
          </div>
        </CardFooter>
      </Card>
    </Col>
  </ForwardRef>
  <ForwardRef
    noGutters={false}
  >
    <Col
      className="Result__redirect"
    >
      <Button
        active={false}
        disabled={false}
        onClick={[Function]}
        type="button"
        variant="primary"
      >
        Back to Home Page
      </Button>
    </Col>
  </ForwardRef>
</Container>
`;

exports[`Result When the transfer process is success When isToEverynet should render correspondingly 1`] = `
<Container
  className="Result"
  fluid={false}
>
  <ForwardRef
    noGutters={false}
  >
    <Col
      className="Result__content"
    >
      <Card
        body={false}
      >
        <CardTitle>
          <div
            className="text-center Result__content__title"
          >
            <i
              className="fas fa-check-circle"
            />
          </div>
        </CardTitle>
        <CardBody
          className="p-0"
        >
          <div
            className="Result__content__body"
          >
            <Container
              fluid={false}
            >
              <ForwardRef
                noGutters={false}
              >
                <Col
                  className="Result__content__body__amount"
                >
                  <h1>
                    1,000 EVRY
                  </h1>
                </Col>
              </ForwardRef>
              <ForwardRef
                noGutters={false}
              >
                <Col
                  className="Result__content__body__status"
                >
                  <h4>
                    TRANSACTION CONFIRMED
                  </h4>
                </Col>
              </ForwardRef>
              <ForwardRef
                noGutters={false}
              >
                <Col
                  className="Result__content__body__account"
                  sm={5}
                  xs={12}
                >
                  <div
                    className="text-left"
                  >
                    <span
                      className="label--dark"
                    >
                      From
                    </span>
                    <Card
                      body={false}
                    >
                      <CardBody>
                        <Container
                          fluid={false}
                        >
                          <ForwardRef
                            noGutters={false}
                          >
                            <span>
                              Stellar
                            </span>
                          </ForwardRef>
                          <ForwardRef
                            noGutters={false}
                          >
                            <span
                              className="Result__content__body__account__text"
                            >
                              Account Number: 
                              foo
                            </span>
                          </ForwardRef>
                        </Container>
                      </CardBody>
                    </Card>
                  </div>
                </Col>
                <Col
                  className="Result__content__body__swap"
                >
                  <i
                    className="fas fa-arrow-right"
                  />
                </Col>
                <Col
                  className="Result__content__body__account"
                  sm={5}
                  xs={12}
                >
                  <div
                    className="text-left"
                  >
                    <span
                      className="label--dark"
                    >
                      To
                    </span>
                    <Card
                      body={false}
                    >
                      <CardBody>
                        <Container
                          fluid={false}
                        >
                          <ForwardRef
                            noGutters={false}
                          >
                            <span>
                              Evrynet
                            </span>
                          </ForwardRef>
                          <ForwardRef
                            noGutters={false}
                          >
                            <span
                              className="Result__content__body__account__text"
                            >
                              Account Number: 
                              bar
                            </span>
                          </ForwardRef>
                        </Container>
                      </CardBody>
                    </Card>
                  </div>
                </Col>
              </ForwardRef>
            </Container>
          </div>
        </CardBody>
        <CardFooter>
          <div
            className="Result__content__footer"
          >
            <Container
              fluid={false}
            >
              <ForwardRef
                noGutters={false}
              >
                <Col
                  className="Result__content__footer__content"
                  xs={6}
                >
                  <Card
                    body={false}
                  >
                    <CardBody
                      className="Result__content__footer__content__body"
                    >
                      Stellar Transaction hash:
                      foo
                    </CardBody>
                  </Card>
                </Col>
                <Col
                  className="Result__content__footer__content"
                  xs={6}
                >
                  <Card
                    body={false}
                  >
                    <CardBody
                      className="Result__content__footer__content__body"
                    >
                      <span>
                        Evrynet Transaction hash: 
                      </span>
                      <span>
                        bar
                      </span>
                    </CardBody>
                  </Card>
                </Col>
              </ForwardRef>
            </Container>
          </div>
        </CardFooter>
      </Card>
    </Col>
  </ForwardRef>
  <ForwardRef
    noGutters={false}
  >
    <Col
      className="Result__redirect"
    >
      <Button
        active={false}
        disabled={false}
        onClick={[Function]}
        type="button"
        variant="primary"
      >
        Back to Home Page
      </Button>
    </Col>
  </ForwardRef>
</Container>
`;

exports[`Result When the transfer process is success When isToStellar should render correspondingly 1`] = `
<Container
  className="Result"
  fluid={false}
>
  <ForwardRef
    noGutters={false}
  >
    <Col
      className="Result__content"
    >
      <Card
        body={false}
      >
        <CardTitle>
          <div
            className="text-center Result__content__title"
          >
            <i
              className="fas fa-check-circle"
            />
          </div>
        </CardTitle>
        <CardBody
          className="p-0"
        >
          <div
            className="Result__content__body"
          >
            <Container
              fluid={false}
            >
              <ForwardRef
                noGutters={false}
              >
                <Col
                  className="Result__content__body__amount"
                >
                  <h1>
                    1,000 EVRY
                  </h1>
                </Col>
              </ForwardRef>
              <ForwardRef
                noGutters={false}
              >
                <Col
                  className="Result__content__body__status"
                >
                  <h4>
                    TRANSACTION CONFIRMED
                  </h4>
                </Col>
              </ForwardRef>
              <ForwardRef
                noGutters={false}
              >
                <Col
                  className="Result__content__body__account"
                  sm={5}
                  xs={12}
                >
                  <div
                    className="text-left"
                  >
                    <span
                      className="label--dark"
                    >
                      From
                    </span>
                    <Card
                      body={false}
                    >
                      <CardBody>
                        <Container
                          fluid={false}
                        >
                          <ForwardRef
                            noGutters={false}
                          >
                            <span>
                              Stellar
                            </span>
                          </ForwardRef>
                          <ForwardRef
                            noGutters={false}
                          >
                            <span
                              className="Result__content__body__account__text"
                            >
                              Account Number: 
                              foo
                            </span>
                          </ForwardRef>
                        </Container>
                      </CardBody>
                    </Card>
                  </div>
                </Col>
                <Col
                  className="Result__content__body__swap"
                >
                  <i
                    className="fas fa-arrow-right"
                  />
                </Col>
                <Col
                  className="Result__content__body__account"
                  sm={5}
                  xs={12}
                >
                  <div
                    className="text-left"
                  >
                    <span
                      className="label--dark"
                    >
                      To
                    </span>
                    <Card
                      body={false}
                    >
                      <CardBody>
                        <Container
                          fluid={false}
                        >
                          <ForwardRef
                            noGutters={false}
                          >
                            <span>
                              Evrynet
                            </span>
                          </ForwardRef>
                          <ForwardRef
                            noGutters={false}
                          >
                            <span
                              className="Result__content__body__account__text"
                            >
                              Account Number: 
                              bar
                            </span>
                          </ForwardRef>
                        </Container>
                      </CardBody>
                    </Card>
                  </div>
                </Col>
              </ForwardRef>
            </Container>
          </div>
        </CardBody>
        <CardFooter>
          <div
            className="Result__content__footer"
          >
            <Container
              fluid={false}
            >
              <ForwardRef
                noGutters={false}
              >
                <Col
                  className="Result__content__footer__content"
                  xs={6}
                >
                  <Card
                    body={false}
                  >
                    <CardBody
                      className="Result__content__footer__content__body"
                    >
                      Stellar Transaction hash:
                      foo
                    </CardBody>
                  </Card>
                </Col>
                <Col
                  className="Result__content__footer__content"
                  xs={6}
                >
                  <Card
                    body={false}
                  >
                    <CardBody
                      className="Result__content__footer__content__body"
                    >
                      <span>
                        Evrynet Transaction hash: 
                      </span>
                      <span>
                        bar
                      </span>
                    </CardBody>
                  </Card>
                </Col>
              </ForwardRef>
            </Container>
          </div>
        </CardFooter>
      </Card>
    </Col>
  </ForwardRef>
  <ForwardRef
    noGutters={false}
  >
    <Col
      className="Result__redirect"
    >
      <Button
        active={false}
        disabled={false}
        onClick={[Function]}
        type="button"
        variant="primary"
      >
        Back to Home Page
      </Button>
    </Col>
  </ForwardRef>
</Container>
`;
